lib/logitech_receiver/hidpp10.py

```diff
@@ -223,18 +223,23 @@ def get_firmware(device):
 
 	firmware = [None, None, None]
 
+	#workarounds for EX100, which doesnt support report firmware
+	fw=_FirmwareInfo(FIRMWARE_KIND.Firmware, '', 'unknown', None)
+	firmware[0]=fw
+
 	reply = read_register(device, REGISTERS.firmware, 0x01)
 	if not reply:
 		# won't be able to read any of it now...
-		return
+		# return unknown for firmware
+		return tuple(f for f in firmware if f)
 
 	fw_version = _strhex(reply[1:3])
 	fw_version = '%s.%s' % (fw_version[0:2], fw_version[2:4])
 	reply = read_register(device, REGISTERS.firmware, 0x02)
 	if reply:
 		fw_version += '.B' + _strhex(reply[1:3])
-	fw = _FirmwareInfo(FIRMWARE_KIND.Firmware, '', fw_version, None)
-	firmware[0] = fw
+		fw = _FirmwareInfo(FIRMWARE_KIND.Firmware, '', fw_version, None)
```

Comment:
Why indent this? If registers not read in line 230, code would return on line 234 never reaching here. If this is a bug fix it would come in a separate commit, if it has no purpose please revert.

Reply:
Indeed, if line 230 succeeds, but line 238 fails, then the firmware version would completely be ignored. Should probably try to report at least part of it.

Reply:
Yes, again rudiment of hacking. I was try to be in style with lines 244-256, but not think about what even part of firmware better than just unknown. I will revert that part.